Fix windows shell provisioner

patux opened the pull request:
This patch fixes the windows shell provisioner,

Issue: https://github.com/WinRb/vagrant-windows/issues/38

sneal commented on lib/vagrant-windows/communication/winrmcommunicator.rb:
```diff
@@ -115,11 +115,11 @@ def upload(from, to)
           $base64_string = Get-Content \"#{file_name}\"
           $bytes  = [System.Convert]::FromBase64String($base64_string) 
-          $new_file = [System.IO.Path]::GetFullPath(\"#{to}\")
+          $new_file = [System.IO.Path]::GetFullPath(\"#{to}.ps1\")
```
The upload method shouldn't be specific to PowerShell scripts.

patux replied:
Right know there is no easy way on windows to tell that a file should run with a specific interpreter like #!/usr/bin/env {interpreter} on linux, If you knew a different way would  be nice to know it

sneal replied:
Adding ".ps1" to any file upload will break other parts of the system. For example, the Chef provisioner uploads a file called "solo.rb", it should not be renamed "solo.rb.ps1".

DEBUG winrmcommunicator: Uploading: /var/folders/z0/83wgqhhj04d3wmk5wrggk5900000gn/T/vagrant20130515-99727-6lta1z to /tmp/vagrant-chef-1/solo.rb

If the file being uploaded needs a "ps1" extension then the provisioner needs to add that before calling upload.

patux replied:
Yeah, make sense, I'll fix that

sneal commented on lib/vagrant-windows/monkey_patches/provisioner.rb:
```diff
@@ -0,0 +1,47 @@
+  module Shell
+      class Provisioner < Vagrant.plugin("2", :provisioner)
+
+        # This patch is needed until Vagrant supports Puppet on Windows guests
```
comment refers to puppet

patux replied:
Oh great I'll remove that line

sneal commented:
Thanks for the pull request. Do you know if this breaks provisioning outside PowerShell from the regular command prompt? I wonder if we'd better off creating a new provisioner type rather than monkey patching, thoughts?

patux commented:
Hi Shawn, not that I knew it, I use the shell provisioner on linux/windows and puppet provisioner and it worked.

I havent test chef.

Creating a new provisioner type instead of monkey patching would be better, that way it would support more than one scripting language. 

#{to} variabale default is /tmp/vagrant-shell 

calling such script directly from powershell would not work, 
AuthorizationManager check failed.
At line:1 char:24
- /tmp/vagrant-shell2.ps1 <<<<
  - CategoryInfo          : NotSpecified: (:) [], PSSecurityException
  - FullyQualifiedErrorId : RuntimeException

and the methods to send parameters to powershell and other commands are siglthy different.. 

I'll continue to investigate on better ways to execute scripts on windows.. 

Right now this works :P 

Let me know if this works for u

patux commented:
Closing this pull request will send another

gorozco1 committed:
Monkey patching shell provisioner for windows

This commit adds monkey_patches to make shell provisioner to work
on windows.

Fix mkdir parameter
mkdir parameter did not work properly

gorozco1 committed:
Bump version to 1.0.3

patux commented:
Another pull request with the fixes for this is sent on https://github.com/WinRb/vagrant-windows/pull/45
